### ISSUE TITLE
Geometry and Mesh information lost when creating the xacro file

### ISSUE THREAD

**Issue by dakuhn** [SammyRamone/sw2urdf_ros2#2]
I have the problem that the tags with the information "visual" and "collision" completely disappear during most conversions from urdf to xarco format.

Is there a reason why this can happen?

URDF
![image](https://github.com/SammyRamone/sw2urdf_ros2/assets/5545470/b86812ca-5dd2-44e8-a274-130396b768fe)

Xacro
![image](https://github.com/SammyRamone/sw2urdf_ros2/assets/5545470/1f2273ee-71f6-481f-b9f9-b64a9ae209da)

**Comment by SammyRamone** [comment]
The script checks if the mesh file for a link exists and deletes all <visual> and <collision> tags for links if this is not the case. I used this to create some links that are purely for tf but have no mesh themselves (typical examples are stuff like base_link, tool0, ...). It might be that there is something going wrong there.
The corresponding code is in these lines https://github.com/SammyRamone/sw2urdf_ros2/blob/69d7443494238a207857b51645736b07c6e63c31/conversion_urdf_ros_2_ros2.py#L152-L164
Basically it goes through all stl files it can find and checks which links are using these. The corresponding links keep their visual and collision tags and all others are removed.
First, you can check if `stl_files` array contains any members. If not your stl files are not correctly found for some reason.

**Comment by dakuhn** [comment]
Thanks for your help, 

I did uncomment the lines and found that 'visual' and 'collision' apears twice, e.g. visual/visual
![image](https://github.com/SammyRamone/sw2urdf_ros2/assets/5545470/60877d38-dbf2-4ae9-a063-eeb9d10ca851)

That could be the reason:
![image](https://github.com/SammyRamone/sw2urdf_ros2/assets/5545470/26dc46a2-dceb-4ec2-b32f-876de7dd10e2)
Here line 118 and line 124 seems to be the same.

Thankfull for any suggestion.

**Comment by SammyRamone** [comment]
Ah correct this line is duplicated somehow. It probably never triggered any issues because my source and target URDF names where different. I removed the duplicated line in the last commit.
This wrongly changed names probably lead to the STL files not being found and then being removed later. Please try again with the newest version if it works.

**Comment by dakuhn** [comment]
Have tested it right now.
It seem to work properly!
Thank you very much!

Another small suggestion for improvement.
At the beginning we could search for the Solidworks output folder, and if it doesn't exist, we could leave the script directly.

**Comment by SammyRamone** [comment]
> Another small suggestion for improvement. At the beginning we could search for the Solidworks output folder, and if it doesn't exist, we could leave the script directly.

Good suggestion, I added it in the last commit.

**Comment by dakuhn** [comment]
> > Another small suggestion for improvement. At the beginning we could search for the Solidworks output folder, and if it doesn't exist, we could leave the script directly.
> 
> Good suggestion, I added it in the last commit.

Thank you very much!
